client/my-sites/add-ons/hooks/use-add-ons.ts

```diff
@@ -140,6 +153,9 @@ const useAddOns = ( siteId?: number ): ( AddOnMeta | null )[] => {
 			}
 		}
 
+		// Determine which Stats Add-On to show based on the site's commercial classification.
+		const isSiteMarkedCommercial = getSiteOption( state, siteId, 'is_commercial' );
```

Comment:
The page flickers on load, but it seems not related to the PR.


https://github.com/Automattic/wp-calypso/assets/1425433/46866ea2-a8cc-4619-943d-0c7c657fae7d